python_modules/dagster/dagster/core/execution/context/hook.py

```diff
@@ -145,6 +145,9 @@ def __init__(
         resources: Dict[str, Any],
         mode_def: Optional[ModeDefinition],
         solid: Optional[Union[SolidDefinition, PendingNodeInvocation]],
+        pipeline_name: Optional[str],
+        run_id: Optional[str],
+        step_key: Optional[str],
```

Comment:
Here's how hook invocation works:
- When `build_hook_context` is called, an `UnboundHookContext` is constructed.
- When that hook context is passed to the invocation of a hook, it is foisted into a `BoundHookContext` object, which has access to the hook's attributes.

As a result, it's necessary to carry forward attributes from `UnboundHookContext` to `BoundHookContext`. You can see where this happens here https://github.com/dagster-io/dagster/blob/master/python_modules/dagster/dagster/core/definitions/hook_invocation.py#L28.

Additionally, these attributes will need to be added to the default `UnboundHookContext`, created here:
https://github.com/dagster-io/dagster/blob/master/python_modules/dagster/dagster/core/definitions/hook_invocation.py#L17.